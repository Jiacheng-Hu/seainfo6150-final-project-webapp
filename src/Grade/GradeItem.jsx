import React from "react";

const GradeItem = props => {
    
    return (
        <tr>
            <td>{props.gradeItem.id}</td>
            <td>{props.gradeItem.name}</td>
            <td>{props.gradeItem.credit}</td>
            <td>{props.gradeItem.year}</td>
            <td>{props.gradeItem.semester}</td>
            <td>{props.gradeItem.grade}</td>
        </tr>
    );
};

export default GradeItem;