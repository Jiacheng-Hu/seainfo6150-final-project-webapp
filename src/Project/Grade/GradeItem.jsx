import React, {useState} from "react";
import GradeToggleButton from "./GradeToggleButton";

const GradeItem = props => {

    const[buttonText, setButtonText] = useState("Show More");

    function onClick(){
        if(buttonText === "Show More"){
            setButtonText("Show Less");
        }else{
            setButtonText("Show More");
        }
    }

    return (
        <tr>
            <td>{props.gradeItem.id}</td>
            <td>{props.gradeItem.name}</td>
            <td>{props.gradeItem.credit}</td>
            <td>{props.gradeItem.year}</td>
            <td>{props.gradeItem.semester}</td>
            <td>
                <div>total: {props.gradeItem.grade}</div>
                {
                    buttonText === "Show Less" ? 
                    <>
                    <div>attendance: {props.gradeItem.attendance}</div>
                    <div>assignment: {props.gradeItem.assignment}</div>
                    <div>exam: {props.gradeItem.exam}</div>
                    </> : null
                }
                <GradeToggleButton onClick={onClick} buttonText={buttonText}/>
            </td>
        </tr>
    )
}

export default GradeItem;