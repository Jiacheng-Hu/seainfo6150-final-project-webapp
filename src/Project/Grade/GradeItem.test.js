import React from "react";
import GradeItem from "./GradeItem";

const gradeItem = {
    slug: "170201-program-structure-and-algorithms",
    id: "170201",
    name: "Program Structure and Algorithms",
    credit: "4",
    year: "2017",
    semester: "Fall",
    grade: "A",
    attendance: "A",
    assignment: "A",
    exam: "A"
};

describe("GradeItem tests", () => {
    it("renders correctly", () => {
        const {container} = render(<GradeItem gradeItem={gradeItem}/>);
        expect(container).toMatchSnapshot();
    });
    it("renders correctly after button is clicked", () => {
        const {container, getByText} = render(<GradeItem gradeItem={gradeItem}/>);
        const buttonShowMore = getByText("Show More");
        fireEvent.click(buttonShowMore);
        expect(container).toMatchSnapshot();
        const buttonShowLess = getByText("Show Less");
        fireEvent.click(buttonShowLess);
        expect(container).toMatchSnapshot();
    });
});