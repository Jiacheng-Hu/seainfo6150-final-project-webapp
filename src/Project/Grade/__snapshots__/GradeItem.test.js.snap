// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradeItem tests renders correctly 1`] = `
<div>
  <tr>
    <td>
      170201
    </td>
    <td>
      Program Structure and Algorithms
    </td>
    <td>
      4
    </td>
    <td>
      2017
    </td>
    <td>
      Fall
    </td>
    <td>
      <div>
        total: 
        A
      </div>
      <button
        class="btn btn-default  btn-sm"
      >
        Show More
      </button>
    </td>
  </tr>
</div>
`;

exports[`GradeItem tests renders correctly after button is clicked 1`] = `
<div>
  <tr>
    <td>
      170201
    </td>
    <td>
      Program Structure and Algorithms
    </td>
    <td>
      4
    </td>
    <td>
      2017
    </td>
    <td>
      Fall
    </td>
    <td>
      <div>
        total: 
        A
      </div>
      <div>
        attendance: 
        A
      </div>
      <div>
        assignment: 
        A
      </div>
      <div>
        exam: 
        A
      </div>
      <button
        class="btn btn-default  btn-sm"
      >
        Show Less
      </button>
    </td>
  </tr>
</div>
`;

exports[`GradeItem tests renders correctly after button is clicked 2`] = `
<div>
  <tr>
    <td>
      170201
    </td>
    <td>
      Program Structure and Algorithms
    </td>
    <td>
      4
    </td>
    <td>
      2017
    </td>
    <td>
      Fall
    </td>
    <td>
      <div>
        total: 
        A
      </div>
      <button
        class="btn btn-default  btn-sm"
      >
        Show More
      </button>
    </td>
  </tr>
</div>
`;
